### PR TITLE
Organized some headers and removed duplicate declarations.

### PR DESCRIPTION
**Pull request by PircDef**
I also added some includes because some of the code depended on tss2 includes to bring in others libraries. I tried to minimize that dependence.

### CODE REVIEW

File: tpm2/include/tpm/formatting_tools.h [hunk 1]
```diff
@@ -8,9 +8,10 @@
 #ifndef FORMATTING_TOOLS_H
 #define FORMATTING_TOOLS_H
 
-#include <stdlib.h>
 #include <tss2/tss2_sys.h>
```
Comment (b-carr) [inline tpm2/include/tpm/formatting_tools.h:11]: Reorder these: standard includes followed by non-standard (but system installed) includes followed by project includes.

File: tpm2/include/tpm/tpm2_interface.h [hunk 2]
```diff
@@ -10,6 +10,8 @@
 
 #include <tss2/tss2_sys.h>
```
Comment (b-carr) [inline tpm2/include/tpm/tpm2_interface.h:11]: Same suggestion here on the order of the includes.

**Review by b-carr — COMMENTED** [review]
Rather than comment every place I'll make this comment here.

I'd prefer we order the includes
(1) Standard includes
(2) Non-standard includes not from our project
(3) Includes from our project.

**Comment by PircDef** [comment]
I ordered the headers like this:

1. Corresponding header file
2. Standard includes
3. 3rd party includes
4. Headers from kmyth

Each is alphabetically sorted (starting with the directory where applicable)

**Pull request merged**